### ISSUE TITLE
Rethink admin table filter UX and implementation

### ISSUE THREAD

**Issue by tvdeyen** [solidusio/solidus#5945]
There are several flaws with the new admin table filter and search implementation.

1. One can only choose one filter at the time
2. The fulltext search submits on key stroke
3. Scope selection disappear when a filter or search is happening
4. You cannot see a filter being enabled without opening a filter dropdown again

Most of that because there is no submit button, lacking any semantics of this form. This is why the code needs to be overly complex, which makes it very hard to maintain or to fix bugs and contribute.

This makes it necessary to run all admin feature specs in full JS enabled browsers, making them slow and flaky. As we are all aware of (nearly every single PR has flaky admin specs).

I like JS to enhance UX, but for simple things like search forms we need to make sure it works without it. Also we have Turbo, wich make forms submits very fast. Stimulus is not necessary for this at all.

Since we are currently laying the foundation for the new admin and encourage the community to contribute, we need to give good examples on how we want to build a maintainable admin interface. The current admin does not give such a good example at the moment.

I am a bit alarming here as we approach to finalize the admin and we as the core team will have to maintain this code for a very long time. Which I am not sure I am willing to.

**Comment by tvdeyen** [comment]
@MadelineCollier @solidusio/core-team Can we address this soonish?

**Comment by kennyadsl** [comment]
@tvdeyen Agree! The figma files are just a reference and should be evaluated with the trade-off of development effort and complexity. Thanks for flagging!

**Comment by MadelineCollier** [comment]
Seems like a sensible edit @tvdeyen, mind if I tackle this after I close out the `store_credit` flow? Apologies, I have had a few weeks off work but back at it this week.

**Comment by tvdeyen** [comment]
Sure @MadelineCollier whenever it works for you. Mostly I wanted to get your thoughts on this topic, but I guess we have an agreement. I am happy. Thanks.
